### DISCUSSION TITLE
how is Recently Played calculated?

### DISCUSSION THREAD

**Discussion by ghost** [navidrome/navidrome#2762] [Q&A] [answered]
I considered filing an issue but I thought I'd ask here first. The Recently Played albums list seems to be generated in some way that I don't understand. I would have expected that any time I play any song from any album, that album would go onto the list; but that is not at all the case. I've played dozens, maybe hundreds of songs from dozens of different albums but there are only three Recently Played albums and I haven't played anything _recently_ from any of them.

The same is true of Most Played (it is the same mysterious list of three as Recently Played).

**Comment by metalheim** [comment, marked as answer]
Navidrome WebUI submits a "play" once a song is played for more than 50% duration.

If you're using external players/clients, it might be handled differently. Some players don't report any plays back to navidrome unless you specifically enable it (many players have an option called "enable scrobbling" - or something similarly-named)

An Albums playCount and lastPlayed value are always derived from it's songs:
- If you play 3 songs from an album, the playCount of the album will be increased by 3
- The album will inherit the largest (latest) lastPlayed value of any song on the album

**Reply by ghost** [reply]
Thanks, that's very comprehensive and should be enough for me to be going on with!

**Reply by ghost** [reply]
Just to follow up: interestingly, this works for my client (iSub): just turning on scrobbling, even though I don't know what scrobbling is and have no `last.fm` identity, causes Recently Played to start working as you describe; but it does _not_ work for Jukebox mode. I would expect that, since Navidrome itself is doing the playing, it would be up to Navidrome to report plays to itself.
